dump-pe/main.cpp

```diff
@@ -272,15 +272,39 @@ int printSecs(void *N,
   return 0;
 }
 
+int printTls(void *N, VA funcAddr) {
+  static_cast<void>(N);
+  auto address = static_cast<std::uint64_t>(funcAddr);
+  std::cout << "TLS: 0x";
+  std::cout << to_string<decltype(address)>(address, std::hex);
+  std::cout << "\n";
+  return 0;
+}
+
+int printTlsWithBinary(void *N, VA funcAddr) {
+  parsed_pe *p = static_cast<parsed_pe*>(N);
+  auto address = static_cast<std::uint64_t>(funcAddr);
+  std::cout << "First 8 bytes from tls callback (0x";
+  std::cout << to_string<VA>(address, std::hex);
+  std::cout << "):" << "\n";
+  for (std::size_t i = 0; i < 8; i++) {
+    std::uint8_t b;
+    ReadByteAtVA(p, i + address, b);
```

Comment:
This call can fail, so we should handle that case (either by ending this particular TLS printout, or maybe by terminating `pe-parse` entirely?)

Reply:
There are the use cases where you want the raw values (pe parser) and the use cases where you want a starting point for an emulator or debugger (pe loader). My take is that `pe-dump` is the former and is probably better suited to printing an appropriate error message saying the address is invalid.

I already have a small change in my local pe-dump that will print `First 8 bytes from tls callback (0x00400000): ERR ERR ERR ERR ERR ERR ERR ERR` if `ReadByteAtVA()` fails. Does this seem substantial enough?